src/lib/viewers/image/__tests__/ImageViewer-test.js

```diff
@@ -576,4 +606,59 @@ describe('lib/viewers/image/ImageViewer', () => {
             expect(image.imageEl.src).toBe(url);
         });
     });
+
+    describe('getViewportDimensions', () => {
+        test('should return width and height', () => {
+            image.wrapperEl = document.createElement('img');
+            Object.defineProperty(image.wrapperEl, 'clientWidth', { value: 100 });
+            Object.defineProperty(image.wrapperEl, 'clientHeight', { value: 100 });
+
+            const result = image.getViewportDimensions();
+
+            expect(result).toEqual({ width: 70, height: 70 });
+        });
+    });
+
+    describe('handleZoomEvent', () => {
+        beforeEach(() => {
+            image.wrapperEl = document.createElement('img');
+            Object.defineProperty(image.wrapperEl, 'clientWidth', { value: 100 });
+            Object.defineProperty(image.wrapperEl, 'clientHeight', { value: 100 });
+            jest.spyOn(image, 'processAnnotationModeChange');
+        });
+
+        test.each`
```

Comment:
Seems like this test may be combining too many things. Maybe split it out to a test based on `type` to make sure it doesn't get processed and the another test for the combinations of `currentState` and whether the image overflows

Reply:
The `should` param is probably unnecessary. Maybe just update the test description to include the `height` and `width`, otherwise consider breaking out a separate test to test the `currentState` vs `overflow` conditions